// formbody/dist/middleware.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.formbody = void 0;
var util_1 = require("util");
var busboy_1 = __importDefault(require("busboy"));
var formbody_1 = require("./formbody");
var formbody = function (req, res, next) {
    if (req.method.toUpperCase() !== 'POST' || !req.is('multipart/form-data')) {
        return next();
    }
    var busboy = new busboy_1.default({ headers: req.headers });
    var formbody = new formbody_1.Formbody();
    busboy.on('file', function (fieldname, file, filename, encoding, mimetype) {
        if (!filename) {
            return;
        }
        var chunks = [];
        file.on('data', function (chunk) {
            chunks.push(chunk);
        });
        file.on('error', function (err) {
            res.status(500).json({
                code: 'form_filter_error',
            });
        });
        file.on('end', function () {
            var finalBuffer = Buffer.concat(chunks);
            formbody.body.push({
                field: fieldname,
                file: {
                    buffer: finalBuffer,
                    size: finalBuffer.length,
                    filename: filename,
                    mimetype: mimetype,
                },
            });
        });
    });
    busboy.on('field', function (fieldname, val) {
        formbody.body.push({
            field: fieldname,
            value: util_1.inspect(val).replace(/(^')|('$)/g, ''),
        });
    });
    busboy.on('finish', function () {
        req.form = formbody;
        next();
    });
    req.pipe(busboy);
};
exports.formbody = formbody;
